workflows/src/main/java/com/compliance/flows/RejectClaimTemplateSuggestion.java

```diff
@@ -0,0 +1,115 @@
+package com.compliance.flows;
+
+import co.paralleluniverse.fibers.Suspendable;
+import com.compliance.contracts.ClaimTemplateContract;
+import com.compliance.states.ClaimTemplateSuggestion;
+import net.corda.core.contracts.StateAndRef;
+import net.corda.core.contracts.UniqueIdentifier;
+import net.corda.core.flows.*;
+import net.corda.core.identity.Party;
+import net.corda.core.node.NodeInfo;
+import net.corda.core.node.services.Vault;
+import net.corda.core.node.services.vault.QueryCriteria;
+import net.corda.core.transactions.SignedTransaction;
+import net.corda.core.transactions.TransactionBuilder;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class RejectClaimTemplateSuggestion {
+
+    @InitiatingFlow
+    @StartableByRPC
+    public static class RejectClaimTemplateSuggestionInitiator extends FlowLogic<SignedTransaction> {
+
+        @NotNull
+        private final UniqueIdentifier linearId;
+
+        public RejectClaimTemplateSuggestionInitiator(@NotNull UniqueIdentifier linearId) {
+            this.linearId = linearId;
+        }
+
+        @Override
+        @Suspendable
+        public SignedTransaction call() throws FlowException {
+
+            final Party notary = getServiceHub().getNetworkMapCache().getNotaryIdentities().get(0);
+            final TransactionBuilder builder = new TransactionBuilder(notary);
+            try {
+                QueryCriteria inputCriteria = new QueryCriteria.LinearStateQueryCriteria()
+                        .withUuid(Collections.singletonList(UUID.fromString(linearId.toString())))
+                        .withStatus(Vault.StateStatus.UNCONSUMED)
+                        .withRelevancyStatus(Vault.RelevancyStatus.RELEVANT);
+
+                final StateAndRef<ClaimTemplateSuggestion> input = getServiceHub().getVaultService().queryBy(ClaimTemplateSuggestion.class, inputCriteria).getStates().get(0);
+                builder.addInputState(input);
+
+                // Add all parties in the network
+                final List<Party> involvedParties = getServiceHub().getNetworkMapCache().getAllNodes().stream().map(NodeInfo::getLegalIdentities).collect(Collectors.toList()).stream().flatMap(List::stream).collect(Collectors.toList());
```

Comment:
We don't need the other involved parties when rejecting a claim template. We simply need the FSP that issued the suggestion and FSA.

Reply:
I think other FSPs should see which suggestions were rejected in order to know what is accepted and whats not.